features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Operations/Operations.jsx

```diff
@@ -0,0 +1,444 @@
+
+/**
+ * This class defined for operation List
+ */
+class Operations extends React.Component {
+    /**
+     *
+     * @param {*} props the props parameters
+     */
+    constructor(props) {
+        super(props);
+        this.state = {
+            swagger: {},
+            scopes: [],
+            notFound: false,
+            showScopes: false,
+            apiPolicies: [],
+            operationList: null,
+        };
+
+        this.newApi = new Api();
+        this.handleUpdateList = this.handleUpdateList.bind(this);
+        this.toggleAssignScopes = this.toggleAssignScopes.bind(this);
+        this.handleScopeChange = this.handleScopeChange.bind(this);
+        this.updateOperations = this.updateOperations.bind(this);
+        this.handleScopeChangeInSwaggerRoot = this.handleScopeChangeInSwaggerRoot.bind(this);
+    }
+
+    /**
+     *
+     */
+    componentDidMount() {
+        this.setState({
+            operationList: this.props.api.operations,
+            scopes: this.props.api.scopes,
+        });
+        const promisedApi = this.newApi.getSwagger(this.props.api.id);
+        promisedApi
+            .then((response) => {
+                let tempScopes = [];
+                if (response.obj.security && response.obj.security.length !== 0) {
+                    response.obj.security.map((object) => {
+                        if (object.OAuth2Security) {
+                            tempScopes = object.OAuth2Security;
+                        }
+                        return tempScopes;
+                    });
+                }
+                this.setState({ swagger: response.obj, scopes: tempScopes });
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') console.log(error);
+                const { status } = error.status;
+                if (status === 404) {
+                    this.setState({ notFound: true });
+                }
+            });
+        const promisedResPolicies = Api.policies('api');
+        promisedResPolicies
+            .then((policies) => {
+                this.setState({ apiPolicies: policies.obj.list });
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.log(error);
+                }
+                const { status } = error.status;
+                if (status === 404) {
+                    this.setState({ notFound: true });
+                } else if (status === 401) {
+                    doRedirectToLogin();
+                }
+            });
+    }
+
+    /**
+     *
+     * @param {*} event triggered for scope change
+     */
+    handleScopeChange(event) {
+        this.setState({ scopes: event.target.value });
+        this.handleScopeChangeInSwaggerRoot(event.target.value);
+    }
+
+    /**
+     *
+     * @param {*} scopes
+     */
+    handleScopeChangeInSwaggerRoot(scopes) {
+        const { swagger } = this.state.swagger;
```

Comment:
Do we have nested `swagger` object here ?

i:e 
```javascript
this.state.swagger.swagger
```

Reply:
done